### ISSUE TITLE
[BUG] Update bulk uploader datetime validations

### ISSUE THREAD

**Issue by emyl3** [CDCgov/prime-simplereport#8253]
### Description

- Discovered while working on #8225
- The results bulk uploader [date time validation](https://github.com/CDCgov/prime-simplereport/blob/26b324f9f2348ca64b4eb398782eaf4269103c2a/backend/src/main/java/gov/cdc/usds/simplereport/validators/CsvValidatorUtils.java#L100) can be updated to address the following scenarios:
   - extra whitespace (e.g. "10/18/2024 10:00" - passes but with extra whitespaces it fails)
   - Verify that the documentation makes the intended format clear.   
   - preceding whitespace in date time string causes a DateTimeParseException to get thrown inside the CompletableFuture returned by TestResultUploadService#submitResultsToCovidPipeline -> this should show an error to the user
   
### Steps to reproduce
[Steps here](https://github.com/CDCgov/prime-simplereport/pull/8225#issuecomment-2445401198)
- use the csv in the comment above

### Proposed Solution

The frontend will already show an error toast if backend doesn't respond with a 200, so we could make sure to throw 
 a **specific error message** to handle exceptions in any case where:

data fails to parse or get transformed before we send to ReportStream due to trailing white space or white space between the dates. Some examples of when we should throw an error: 

```
"01 / 01 / 2023 11 : 11"
"01/01/2023 11:11 EST  "
" 01/01/2023 11:11 EST "
```

#### Additional Context

please be wary of overlap work on which is meant to throw a generic error message: 
[https://app.zenhub.com/workspaces/simplereport-2025-and-onwards-605b43a6ceb92c000f86279a/issues/gh/cdcgov/prime-simplereport/8496](url)

**Comment by mindyatwork** [comment]
Please address the whitespace issue.

If handling the AM/PM abbreviation is a ~1 point lift, please address that too. If more than that, we should investigate from the logs if users commonly run into this error.

**Comment by jayna-SkylightDigital** [comment]
> [@jayna-SkylightDigital](https://github.com/jayna-SkylightDigital) a quick question on this ticket. Is there a max number of white spaces the user is allowed to input here? For example: `01/01/ 2023 11:11 PT` might be a challenge for our parser to be able to process.

@arinkulshi-skylight I ran this by the team, and @DanielSass's recommendation is that additional spaces included in dates would result in an error:
> if people are putting extra spaces in the middle of their date times, that’s not the api and that should be an error in my opinion.

**Comment by DanielSass** [comment]
Moving back to `Needs requirements` until Mindy is back in office.  I misunderstood the behavior being described and I don't think it constitutes a bug.  Want to make sure we're all on the same page before we close this.

**Comment by mindyatwork** [comment]
A recommendation: 

- any additional whitespace in the middle of the string, ex `01/01/    2023 11:11 PT`, should throw an error at the validation step so that users can fix it.

Question:
- what happens in the case of a true preceding/trailing whitespace ex. ` 01/01/2023 11:11 PT`? Is that what you were describing @mehansen as missing a validation error?

**Comment by mehansen** [comment]
> Question:
> 
> * what happens in the case of a true preceding/trailing whitespace ex. ` 01/01/2023 11:11 PT`? Is that what you were describing [@mehansen](https://github.com/mehansen) as missing a validation error?

@mindyatwork Yes I'm seeing in the logs that we're accepting this in the validation step (so not giving a validation error to the user) but then when we try to parse it to format it to send to RS we are failing so it doesn't get sent and we don't give a success or fail feedback to the user.

**Comment by arinkulshi-skylight** [comment]
So to confirm instead of removing the white space for the user programatically. When should throw an error prompting the user to fix the trailing whitespace.

**Comment by arinkulshi-skylight** [comment]
I have updated the scope of this ticket. Let me know if you would like to make any additional changes.

**Comment by mindyatwork** [comment]
@arinkulshi-skylight lgtm from product side. @DanielSass hbu?

**Comment by mehansen** [comment]
Just want to clarify that (unless I'm mistaken) we don't want a new type of error toast for failing to parse the dates due to whitespace, but rather we would include them in the usual validation feedback format that we use currently for validation errors

**Comment by mindyatwork** [comment]
> Just want to clarify that (unless I'm mistaken) we don't want a new type of error toast for failing to parse the dates due to whitespace, but rather we would include them in the usual validation feedback format that we use currently for validation errors

Yes, that sounds good to me. @mehansen

**Comment by DavidMcClatchey** [comment]
Just looking through this ticket and the comments and wanted to point out that as part of [this issue](https://github.com/CDCgov/prime-simplereport/issues/8496) and [the PR changes in DateTimeUtils here](https://github.com/CDCgov/prime-simplereport/pull/8607/files#diff-088f0816db798902a1772c1a1f48d6394d63ceea491037770c3727419ea84488), valid datetimes with trailing and preceding whitespaces would pass validation, but whitespaces in the middle of datetimes would not. 

After discussing with a couple of people in pairing sessions, this seemed like the way to go since it feels like unnecessary friction and manual effort to ask the user to remove trailing/preceding whitespaces in an otherwise valid date.

**Comment by mindyatwork** [comment]
@DavidMcClatchey 's proposal sounds good to me.
